packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -721,6 +740,50 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   * Adds a volume to the Service.
+   */
+  public addVolume(volume: ServiceManagedVolume) {
+    // only one ServiceVolumeConfiguration can be specified.
+    if (this.volumes.length >= 1) {
+      throw new Error('Invalid VolumeConfiguration. Only one volume can be configured at launch.');
```

Comment:
It is good to count the number and occur an error in the `renderVolumes` instead of making an error the second time it is called in the `addVolume` method.
Since the `renderVolumes` method is called in a lazy method (643 line), which allows for lazy evaluation, it would be kind to indicate with `got: ${}` in the error message how many volumes were finally specified.